### PR TITLE
Fix pagination tests

### PR DESCRIPTION
**Pull request by garrensmith**
Removes the excludes for all databases except SqlServer. I had to add in extra results because most sql engines handle ordering of null in a different way. 

This fixes https://github.com/prisma/prisma-engines/issues/2296

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -245,6 +255,7 @@ services:
     environment:
       MONGO_INITDB_ROOT_USERNAME: "prisma"
       MONGO_INITDB_ROOT_PASSWORD: "prisma"
+      INIT_WAIT_SEC: 10
```
Comment (dpetrick) [inline docker-compose.yml:258]: Nitpick: Can you try lowering it? Since this applies to all local devs, 10s is a long time to switch dev dbs.

Comment (dpetrick) [reply docker-compose.yml:258]: Until you hit problems again that is.

File: query-engine/connector-test-kit-rs/query-engine-tests/src/utils/querying.rs [hunk 2]
```diff
@@ -7,16 +7,32 @@ macro_rules! assert_query {
 }
 
 #[macro_export]
-macro_rules! assert_query_many {
-    ($runner:expr, $q:expr, $potential_results:expr) => {
+macro_rules! connector_results {
```
Comment (dpetrick) [inline query-engine/connector-test-kit-rs/query-engine-tests/src/utils/querying.rs:10]: I think I proposed this name initially when I wrote you, but that was more or less a placeholder as naming this is hard 😄 `diverging_results` maybe? `possible_results`? something along those lines perhaps.

File: query-engine/connector-test-kit-rs/query-engine-tests/tests/queries/aggregation/group_by_having.rs [hunk 3]
```diff
@@ -54,12 +54,9 @@ mod aggregation_group_by_having {
                 _sum { int }
               }
             }"#,
-            vec![
-                r#"{"data":{"groupByTestModel":[{"string":"group1","int":5,"_count":{"_all":1},"_sum":{"int":5}},{"string":"group2","int":5,"_count":{"_all":1},"_sum":{"int":5}}]}}"#, // SQL
-                r#"{"data":{"groupByTestModel":[{"string":"group2","int":5,"_count":{"_all":1},"_sum":{"int":5}},{"string":"group1","int":5,"_count":{"_all":1},"_sum":{"int":5}}]}}"# // Mongo
-            ]
+            [SqlServer, Postgres, MySql, Sqlite, MongoDb] => r#"{"data":{"groupByTestModel":[{"string":"group1","int":5,"_count":{"_all":1},"_sum":{"int":5}},{"string":"group2","int":5,"_count":{"_all":1},"_sum":{"int":5}}]}}"#,
```
Comment (dpetrick) [inline query-engine/connector-test-kit-rs/query-engine-tests/tests/queries/aggregation/group_by_having.rs:57]: MongoDB is in both branches?

File: query-engine/connector-test-kit-rs/query-engine-tests/tests/queries/filters/insensitive_filters.rs [hunk 4]
```diff
@@ -83,24 +81,20 @@ mod insensitive {
...
-            vec![
-                r#"{"data":{"findManyTestModel":[{"str":"æ"},{"str":"Æ"},{"str":"bar"},{"str":"aÆB"},{"str":"aæB"}]}}"#, // Mongo
-                r#"{"data":{"findManyTestModel":[{"str":"æ"},{"str":"Æ"},{"str":"bar"},{"str":"aÆB"},{"str":"AÆB"},{"str":"aæB"},{"str":"aB"}]}}"#, // Postgres
-                r#"{"data":{"findManyTestModel":[{"str":"æ"},{"str":"Æ"},{"str":"bar"},{"str":"aÆB"},{"str":"AÆB"},{"str":"aæB"}]}}"#, // Cockroach, https://github.com/cockroachdb/cockroach/issues/71313
-            ]
+        connector_results!(
+          &runner,
+          r#"query { findManyTestModel(where: { str: { gte: "aÆB", mode: insensitive } }) { str }}"#,
+          [MongoDb] => r#"{"data":{"findManyTestModel":[{"str":"æ"},{"str":"Æ"},{"str":"bar"},{"str":"aÆB"},{"str":"aæB"}]}}"#,
+          [Sqlite, SqlServer, MySql, Postgres] => r#"{"data":{"findManyTestModel":[{"str":"æ"},{"str":"Æ"},{"str":"bar"},{"str":"aÆB"},{"str":"AÆB"},{"str":"aæB"},{"str":"aB"}]}}"#, // Postgres
```
Comment (dpetrick) [inline query-engine/connector-test-kit-rs/query-engine-tests/tests/queries/filters/insensitive_filters.rs:88]: One cool thing would be even version differentiation. We don't need it now, just thinking out loud. Like `MySql(5.7)`.

File: query-engine/connector-test-kit-rs/query-engine-tests/tests/queries/order_and_pagination/nested_multi_order_pagination.rs [hunk 5]
```diff
@@ -244,22 +239,18 @@ mod paging_one2m_unstable_order {
...
+              findManyTestModel {
                 id
+                related(cursor: { id: 3 }, orderBy: [{ fieldA: desc }, { fieldB: asc }, { fieldC: asc }, { fieldD: desc }]) {
+                  id
+                }
               }
-            }
-          }"#,
-            vec![
-                r#"{"data":{"findManyTestModel":[{"id":1,"related":[]},{"id":2,"related":[{"id":3},{"id":4}]},{"id":3,"related":[]}]}}"#,
```
Comment (dpetrick) [inline query-engine/connector-test-kit-rs/query-engine-tests/tests/queries/order_and_pagination/nested_multi_order_pagination.rs:258]: That is somewhat surprising to me that this is different now.

File: query-engine/connector-test-kit-rs/query-engine-tests/tests/queries/filters/insensitive_filters.rs (hunk 4 above)
Comment (dpetrick) [reply query-engine/connector-test-kit-rs/query-engine-tests/tests/queries/filters/insensitive_filters.rs:88]: Re-reading some of the code: Actually, it seems like that a connector in two branches is basically a version differentiation already, am I correct? In that case the macro semantics in my head and how you solved it are different: I basically thought that the `x => y` syntax should go into a `match { x => y}` (just more complicated probably) that asserts for that connector that result. The way you take it works as well though, so no worries, it's just not immediately clear to me that putting a connector into multiple branches stacks the possible results for that connector.

File: query-engine/connector-test-kit-rs/query-engine-tests/tests/queries/order_and_pagination/order_by_dependent_pagination.rs [hunk 6]
```diff
@@ -286,12 +285,14 @@ mod order_by_dependent_pag {
...
-            vec![
-                r#"{"data":{"findManyModelA":[{"id":2,"b":{"c":{"a":{"id":4}}}},{"id":1,"b":{"c":{"a":{"id":3}}}},{"id":3,"b":null},{"id":4,"b":null}]}}"#,
-                r#"{"data":{"findManyModelA":[{"id":2,"b":{"c":{"a":{"id":4}}}},{"id":1,"b":{"c":{"a":{"id":3}}}}]}}"#,
-            ]
+            [MongoDb, Sqlite, MySql] => r#"{"data":{"findManyModelA":[{"id":2,"b":{"c":{"a":{"id":4}}}},{"id":1,"b":{"c":{"a":{"id":3}}}},{"id":3,"b":null},{"id":4,"b":null}]}}"#,
+            // [MySql] => r#"{"data":{"findManyModelA":[{"id":2,"b":{"c":{"a":{"id":4}}}},{"id":1,"b":{"c":{"a":{"id":3}}}}]}}"#,
+            [Postgres] => r#"{"data":{"findManyModelA":[{"id":3,"b":null},{"id":4,"b":null},{"id":2,"b":{"c":{"a":{"id":4}}}},{"id":1,"b":{"c":{"a":{"id":3}}}}]}}"#
+            // MySql 5
+            // [MySql] => r#"{"data":{"findManyModelA":[{"id":2,"b":{"c":{"a":{"id":4}}}},{"id":1,"b":{"c":{"a":{"id":3}}}},{"id":4,"b":null},{"id":3,"b":null}]}}"#,
+            //MariaDB, MySql8
```
Comment (dpetrick) [inline query-engine/connector-test-kit-rs/query-engine-tests/tests/queries/order_and_pagination/order_by_dependent_pagination.rs:293]: What happened here?

File: query-engine/connector-test-kit-rs/query-engine-tests/tests/queries/order_and_pagination/nested_multi_order_pagination.rs (hunk 5 above)
Comment (dpetrick) [reply query-engine/connector-test-kit-rs/query-engine-tests/tests/queries/order_and_pagination/nested_multi_order_pagination.rs:258]: There's more examples down below. Curious.

File: docker-compose.yml [hunk 7]
```diff
@@ -215,6 +215,16 @@ services:
       - "1433:1433"
     networks:
       - databases
+  azure-edge:
```
[the remark this answers is not in the document]
Comment (garrensmith) [reply docker-compose.yml:218]: Yes. Its not an exact copy of sqlserver. But better than nothing.

File: query-engine/connector-test-kit-rs/query-engine-tests/tests/queries/aggregation/group_by_having.rs (hunk 3 above)
Comment (garrensmith) [reply query-engine/connector-test-kit-rs/query-engine-tests/tests/queries/aggregation/group_by_having.rs:57]: Right. MongoDB 4.2 and Mongo 5 give different results.

File: query-engine/connector-test-kit-rs/query-engine-tests/tests/queries/order_and_pagination/order_by_dependent_pagination.rs (hunk 6 above)
Comment (garrensmith) [reply query-engine/connector-test-kit-rs/query-engine-tests/tests/queries/order_and_pagination/order_by_dependent_pagination.rs:293]: Sorry code clean up that I missed

File: query-engine/connector-test-kit-rs/query-engine-tests/tests/queries/filters/insensitive_filters.rs (hunk 4 above)
Comment (garrensmith) [reply query-engine/connector-test-kit-rs/query-engine-tests/tests/queries/filters/insensitive_filters.rs:88]: Its not quite possible to do that at the moment. I would say we could do that later.

File: query-engine/connector-test-kit-rs/query-engine-tests/tests/queries/order_and_pagination/nested_multi_order_pagination.rs (hunk 5 above)
Comment (garrensmith) [reply query-engine/connector-test-kit-rs/query-engine-tests/tests/queries/order_and_pagination/nested_multi_order_pagination.rs:258]: Yeah. Once I retested, the results were all consistent.

**Review by dpetrick — APPROVED** [review]

**Pull request merged**